tests/unit_tests/plugins/modules/readers/test_readers.py

```diff
@@ -0,0 +1,88 @@
+# # # Distribution Statement A. Approved for public release. Distribution unlimited.
+# # #
+# # # Author:
+# # # Naval Research Laboratory, Marine Meteorology Division
+# # #
+# # # This program is free software: you can redistribute it and/or modify it under
+# # # the terms of the NRLMMD License included with this program. This program is
+# # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+# # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+# # # for more details. If you did not receive the license, for more information see:
+# # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+"""Unit tests on all the readers."""
+import pytest
+from geoips.commandline.log_setup import setup_logging
+from geoips.interfaces import readers
+
+
+LOG = setup_logging()
+
+
+class TestReaders:
+    """Unit tests every reader."""
+
+    available_readers = [
+        ("abi_l2_netcdf"),
+        ("abi_netcdf"),
+        ("ahi_hsd"),
+        ("amsr2_netcdf"),
+        ("amsr2_remss_winds_netcdf"),
+        ("amsub_hdf"),
+        ("amsub_mirs"),
+        ("ascat_uhr_netcdf"),
+        ("atms_hdf5"),
+        ("ewsg_netcdf"),
+        ("geoips_netcdf"),
+        ("gmi_hdf5"),
+        ("imerg_hdf5"),
+        ("mimic_netcdf"),
+        ("modis_hdf4"),
+        ("saphir_hdf5"),
+        ("sar_winds_netcdf"),
+        ("scat_knmi_winds_netcdf"),
+        ("scat_noaa_winds_netcdf"),
+        ("seviri_hrit"),
+        ("sfc_winds_text"),
+        ("smap_remss_winds_netcdf"),
+        ("smos_winds_netcdf"),
+        ("ssmi_binary"),
+        ("ssmis_binary"),
+        ("viirs_netcdf"),
+        ("wfabba_ascii"),
+        ("windsat_idr37_binary"),
+        ("windsat_remss_winds_netcdf"),
+    ]
+
+    def verify_plugin(self, plugin):
+        """Yeild test xarray and parameters."""
+        test_xr = plugin.module.yeild_test_files()
+        test_param = plugin.module.yeild_test_parameters()
+        self.verify_xarray(test_xr, test_param)
```

Comment:
So we could additionally return a variable name to check, and min/max/mean expected value for that variable in yield_test_parameters? Then in "verify_xarray" we could pass in those values to check and make sure they match the expected value?  So you would have to pre-determine what the min/max/mean are, then check those explicit values against what was just read to make sure the reader is still working as expected.

@jsolbrig Is that what you were thinking?

Reply:
Yes, this is exactly what I was thinking. I think this looks great so far!

Reply:
Will address with https://github.com/NRLMMD-GEOIPS/geoips/issues/390